spec/models/vim_performance_state_spec.rb

```diff
@@ -1,5 +1,40 @@
 RSpec.describe VimPerformanceState do
-  describe ".capture_host_sockets" do
+  let(:capture_time) { 5.hours.ago.beginning_of_hour }
+
+  describe ".capture" do
+    it "uses now for Infra" do
+      hardware = FactoryBot.create(:hardware, :cpu_sockets => 2)
+      host = FactoryBot.create(:host, :hardware => hardware)
+      state = VimPerformanceState.capture(host, capture_time)
+
+      expect(state.timestamp).to be_within(1.minute).of(Time.now.beginning_of_hour)
+    end
+
+    it "uses time passed in for containers" do
+      container_image = FactoryBot.create(:container_image)
+      container = FactoryBot.create(:container, :container_image => container_image)
+      state = VimPerformanceState.capture(container_image, capture_time)
+
+      expect(state.timestamp).to be_within(1.minute).of(capture_time)
+    end
+  end
+
+  describe "#capture_assoc_ids" do
+    it "captures runing containers" do
+      container_image = FactoryBot.create(:container_image)
+      # not deleted is captured
+      containers = FactoryBot.create_list(:container, 1, :container_image => container_image)
+      # deleted later is captured
+      containers << FactoryBot.create(:container, :container_image => container_image, :deleted_on => (capture_time + 2.hours))
+      # deleted before is not captured
+      FactoryBot.create(:container, :container_image => container_image, :deleted_on => (capture_time - 2.hours))
+
+      state = VimPerformanceState.capture(container_image, capture_time)
+      expect(state.get_assoc(:containers).map(&:to_i).sort).to eq(containers.map(&:id).sort)        
```

Comment:
```suggestion
      expect(state.get_assoc(:containers)).to match_array(containers)
```

Reply:
this is comparing `to_i` to `id`, so I did need to keep that in there